### ISSUE TITLE
请问attention mask矩阵是上三角矩阵吗？还是类似chatglm1的attention mask矩阵 我看了源码，在推理的时候attention mask 是全一，意味着输入的问题相互都是能看到的，不是decoder only结构了，但是我发现我微调用上三角矩阵效果反而比类似chatglm1的attention mask矩阵效果好，求解答，微调时候的attention mask矩阵

### ISSUE THREAD

**Issue by shyoulala** [THUDM/ChatGLM2-6B#61]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

请问attention mask矩阵是上三角矩阵吗？还是类似chatglm1的attention mask矩阵
我看了源码，在推理的时候attention mask 是全一，意味着输入的问题相互都是能看到的，不是decoder only结构了，但是我发现我微调用上三角矩阵效果反而比类似chatglm1的attention mask矩阵效果好，求解答，微调时候的attention mask矩阵

### Steps To Reproduce

.

### Environment

```markdown
- OS:linux
- Python:3.8
- Transformers:
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by duzx16** [comment]
transformers 里的 attention mask 是表示对应的 token 是否是 padding的，全1 代表没有 padding。实际用的还是下三角矩阵，可以看 get_masks 这个函数
